### PR TITLE
refactor(core): Update `ComponentFixture` behavior when using zoneless scheduler

### PR DESCRIPTION
**Pull request by atscott**
When the zoneless scheduler is provided, we want to update the behavior of `ComponentFixture` to address common issues and painpoints in testing. Developers should never have to call `detectChanges` on a fixture manually. Instead of calling `detectChanges` after performing an action that updates state and requies a template refresh, developers should wait for change detection to run because the update needs to also have notified the scheduler. If this was not the case, the component would not work correctly in the application. Calling `detectChanges` to force an update could hide real bugs.

This commit also updates the zoneless tests to uses `ComponentFixture` instead of manually attaching to the `ApplicationRef` and rewriting a lot of the helpers (`getDebugNode`, `isStable` as a value, `whenStable` as a Promise).

### CODE REVIEW

File: packages/core/testing/src/component_fixture.ts [hunk 1]
```diff
@@ -235,8 +151,145 @@ export class ComponentFixture<T> {
...
+    }
+    return firstValueFrom(this._appRef.isStable.pipe(filter(stable => stable)));
+  }
+
+  override autoDetectChanges(autoDetect?: boolean|undefined): void {
+    throw new Error('Cannot call autoDetectChanges when using change detection scheduling.');
+  }
+}
+
+export class PseudoApplicationComponentFixture<T> extends ComponentFixture<T> {
```
Comment (jessicajaniuk) [inline packages/core/testing/src/component_fixture.ts:197]: Can we have some docs for `ScheduledComponentFixture` and `PseudoApplicationComponentFixture`?

**Review by jessicajaniuk — APPROVED** [review]
LGTM. Just a non-blocking request for some added docs around the two ComponentFixture classes.

reviewed-for: fw-core, fw-testing, public-api,

**Review by AndrewKushnir — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 3ca34e606d9bc11abe42bbb37c5d6a2b74dc17ea.

**Pull request closed without merging**